### PR TITLE
News feed improvements (and bonus storybook things)

### LINKED ISSUE #2737: News feed improvements
The news feed could use a few optimizations:
1. Items should be hidden by default after two weeks.
2. All Items should be in reverse chronological order (currently, they are ordered by experiment)
3. There should be an option to load older posts (see secreenshot), clicking this should load the five previous news items below
4. If a user has loaded all historical news items the load more link should disappear.
5. If a news item is clickable, it should include a GA ping `ec: ExperimentsPage Interactions, ea: click, el: news-item-{id}`

### PR DESCRIPTION
**Pull request by lmorchard**
- Separate news updates into "fresh" and "stale" (i.e. 2-week-old) items

- Only show "fresh" items by default, offer a "show more" button if
  stale items are available

- Reveal only up to 5 stale items when the "show more" button is clicked

- Updates to test & stories

- Ran prettier on some of the edited files

- Move from stories to frontend/stories, dynamically load, and
  rename .js -> .jsx

Fixes #2737

### CODE REVIEW

**Comment by lmorchard** [comment]
Still work-in-progress pending:
- [ ] some questions on #2737 
- [x] needs some button styling,
- [x] and a GA event on item click.

Try using Storybook locally to run this through its paces:
* `npm start` in one shell to build the site and static assets
* `npm run storybook` in another shell to start the Storybook server
* once the above two things settle down, visit http://localhost:6006/ and click on `UpdateList` in left sidebar

<strike>If I'm very lucky, this URL might work after CircleCI goes green:</strike> (nope, that didn't work, try running storybook locally)

File: frontend/stories/app/components/UpdateList-story.jsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import { storiesOf } from '@storybook/react';
+import { action } from '@storybook/addon-actions';
+import { withKnobs, object, boolean } from '@storybook/addon-knobs';
+
+import UpdateList from '../../../src/app/components/UpdateList';
+import LayoutWrapper from '../../../src/app/components/LayoutWrapper';
+
+let time = Date.now();
+
+// quick & dirty utiity to make canned news updates
```
Comment (pdehaan) [inline frontend/stories/app/components/UpdateList-story.jsx]: typo: utility 🚨

**Comment by pdehaan** [comment]
> Move from stories to frontend/stories, dynamically load, and
> rename .js -> .jsx

FWIW, I think renaming to *.jsx will prevent ESLint from checking those files (if that matters).
https://github.com/mozilla/testpilot/blob/fc5ff136d46f12c733a0f9ae59874e6d10ad5f3d/frontend/tasks/scripts.js#L44-L49

File: frontend/tasks/scripts.js [hunk 2]
```diff
@@ -44,7 +44,7 @@ function shouldLint(opt, task) {
 }
 
 gulp.task('scripts-lint', () => {
-  return gulp.src([config.SRC_PATH + '*.js', config.SRC_PATH + '{app,tests}/**/*.js'])
+  return gulp.src('./frontend/{src,tests,stories}/**/*.{js,jsx}')
```
Comment (lmorchard) [inline frontend/tasks/scripts.js:47]: > FWIW, I think renaming to *.jsx will prevent ESLint from checking those files (if that matters).

Turns out we were missing more than *.jsx from ESLint checking

**Comment by lmorchard** [comment]
Okay, I think I'm down to just some outstanding questions (/cc @johngruen):

- [ ] Button styling okay? (I borrowed the "View Past Experiments" styles)
- [ ] Do items appear in reverse chronological order? [This code hasn't been changed in this PR](https://github.com/mozilla/testpilot/blob/master/frontend/src/app/selectors/news.js#L25) and already passed [a unit test](https://github.com/mozilla/testpilot/blob/master/frontend/test/app/selectors/news-test.js#L57) - if things appear out of order, I need to update the unit test.
- [ ] What should things look like if there's no news to be shown at all? (i.e. everything's older than 2 weeks) Our current component just hides itself and the "Latest News" header - the "View More News" button won't even appear.

**Comment by lmorchard** [comment]
Trying a new Storybook URL over here:

http://testpilot-storybook.dev.mozaws.net/pull/2765/index.html

**Comment by johngruen** [comment]
>  Button styling okay? (I borrowed the "View Past Experiments" styles)

Yeah, give it 10px of top margin, and maybe change the string to "Show Past News"

> Do items appear in reverse chronological order? This code hasn't been changed in this PR and already passed a unit test - if things appear out of order, I need to update the unit test.

Yes, this is working correctly already, my mistake

> What should things look like if there's no news to be shown at all? (i.e. everything's older than 2 weeks) Our current component just hides itself and the "Latest News" header - the "View More News" button won't even appear.

I think that's fine for now...We could add a link to a feed page at some point from settings, but that seems to be separate work.

**Review by johngruen — APPROVED** [review]
Added a tiny ui/string tweak, but UX lgtm

**Comment by lmorchard** [comment]
Oof, this needed a string change and a squash before merge :/

**Comment by fzzzy** [comment]
@lmorchard oh no, I looked and it seemed like it was squashed.

**Comment by lmorchard** [comment]
Follow up PR https://github.com/mozilla/testpilot/pull/2815

**Pull request merged**